### PR TITLE
Revamping the Schema Documentation

### PR DESCRIPTION
**Pull request by jenningsanderson**
View the proposed changes here: https://dfhx9f55j8eg5.cloudfront.net/pr/102

1. Bumps to Docusaurus 3.1.0, which fixes the security vulnerabilities
2. Changes sidebar structure to address "common" elements, such as the new names schema.

Broadly, this takes the documentation from the previous "We are introducing the Overture schema, it will be like this" to "This is what the schema is"

### CODE REVIEW

**Comment by vcschapp** [comment]
I created a GitHub issue to track this work. My apologies for creating it after the fact.

https://github.com/OvertureMaps/schema-wg/issues/251

I put somewhat stringent "acceptance criteria" into that issue... Let me know if they make sense:

> 1. [ ] MUST add a sub-page structure into the documentation to cover "common" schema items.
> 2. [ ] MUST refactor the Transportation theme's [Scoped And Rule-Based Properties](https://docs.overturemaps.org/themes/transportation/scoping-rules) to lift it up to fit within the "common" schema sub-pages.
> 3. [ ]  MUST document the new names schema within a sub-page in the "common" schema area.

If these ACs make sense then I think this change only partially accomplishes item (3) but leaves item (1) and (2) incomplete...

-----

What if we structured things a bit more like this:

 > - Overview
>     - Introduction ...
>     - Feature Model
>         - Data Formats: GeoJSON and GeoParquet (no content currently)
>         - Names
>         - Scoped and Rule-Based Properties

Then:

1. The content of "Introduction" would mention that the Overture schema uses a simple feature model and that text would link out to the "Feature Model" page.
2. The content of the "Feature Model" page would have headings about:
    - Simple Features and why we use them
    - Common Feature Properties
        - This would list all the common properties maybe in a table. The entry for names would link to the Names page.
    - Common Concepts
        - This would list all the common concepts we can think of. The only concept I can currently think of is the Scoped and Rule-based properties concept, which would link out to that page.
    - Data Formats. This would have a sentence or two about canonical GeoJSON and GeoParquet and would link out to the sub-page for that item, which will be empty and listed as "under construction".

File: docusaurus/package.json
Comment (vcschapp) [inline docusaurus/package.json:32]: Big bounce! good to see those security issues going away.

**Review by vcschapp — COMMENTED** [review]
Left thoughts on the docs content.

**Comment by jenningsanderson** [comment]
@danabauer — We're considering some big updates to the structure of the schema documentation here. What are your thoughts?

File: docusaurus/docs/overview/feature-model/names.mdx
Comment (vcschapp) [inline docusaurus/docs/overview/feature-model/names.mdx:36]: 1. [ ] Ideally we should explain the link between `rules` and "Scoped and Rule-Based Properties".
2. [ ] Ideally we should give some further explanation of how to build up names rules and what your options are, *e.g.* language, side, LR, *etc.*
3. [ ] I think that the "rule evaluation algorithm" described on the scoped and rule-based properties page may need to be massaged to explain how to resolve name rules. I'm not 100% sure yet, but we should think about it.

File: docusaurus/docs/overview/feature-model/scoping-rules.mdx
Comment (vcschapp) [inline docusaurus/docs/overview/feature-model/scoping-rules.mdx:2]: We maybe should change the title to "Scoping *and* rules"...

Technically, *scoping* and *rules* are two different concepts.

1. All rule-based properties involve scoping.
1. But a property can be scoped, without being rule-based.

File: docusaurus/docs/overview/feature-model/scoping-rules.mdx
Comment (vcschapp) [inline docusaurus/docs/overview/feature-model/scoping-rules.mdx:50]: This paragraph highlights the fact that the page content was originally written from an *exclusively* transportation focus, and probably should be massaged to be more generically-focused.

The problem with this paragraph is it assumes "your brain context = transportation" so it starts talking about segments, which are a transportation context. Maybe in this particular example, what is meant is "linear features"?

Probably the whole page needs to be audited carefully from that perspective.

**Review by vcschapp — COMMENTED** [review]
Thanks @jenningsanderson and @danabauer!

I believe some changes should probably be done *after this PR* to smooth off some rough edges - some example comments are flagged within. But it's my preference to get this one merged and then maybe tackle those changes later, to avoid delaying the important changes.

@danabauer Would it be possible to add some of the comments to your TODO list? If so, I would suggest we approve and merge this one and just let the refinements come as they're ready.

**Comment by jenningsanderson** [comment]
Nice @vcschapp — I agree. Maybe we could get this PR into the minimum: "we're happy if this goes live with the next data release" state, meaning it has: 
- [x] Better intro
- [ ] Describes the common `name` schema
- [ ] Refreshes GERS pages a bit? 

at the minimum, and then we merge it into `dev` (so it's ready to get to pushed to `main`) — and then we can open a new PR with incremental changes?

**Comment by danabauer** [comment]
I'll fix this!

File: docusaurus/docs/overview/feature-model/index.mdx
Comment (jwass) [inline docusaurus/docs/overview/feature-model/index.mdx:51]: This needs to be updated to latest names schema

File: docusaurus/docs/overview/feature-model/index.mdx
Comment (jwass) [inline docusaurus/docs/overview/feature-model/index.mdx:58]:
```
"names": {
  "primary": "Springfield Town Hall"
},
```

**Review by jwass — APPROVED** [review]
Approved subject to recommended change

**Comment by vcschapp** [comment]
Reminder: @TristanDiet-TomTom and @DavidKarlas committed to reviewing this by no later than tomorrow. We also agreed that a "late merge" to `dev` past the freeze date is acceptable here because it is only docs.

**Comment by danabauer** [comment]
> We also agreed that a "late merge" to `dev` past the freeze date is acceptable 

That's great @vcschapp. What is the freeze date?

File: docusaurus/docs/gers/gers.mdx
Comment (jenningsanderson) [inline docusaurus/docs/gers/gers.mdx:19]: @danabauer — I changed the order of these: h3 tile is first

File: docusaurus/docs/overview/feature-model/index.mdx
Comment (jenningsanderson) [inline docusaurus/docs/overview/feature-model/index.mdx:15]: @danabauer — I took out some of the newlines here to make it more readable; do you think this is OK? Do you think we could even go to just 1 line for the coordinates?

**Comment by vcschapp** [comment]
> > We also agreed that a "late merge" to `dev` past the freeze date is acceptable
> 
> That's great @vcschapp. What is the freeze date?

Today technically!

Our current plan (until we think of a better one) is to freeze merges to the schema's `dev` branch two weeks before the release and unfreeze them on the day the release goes out.

File: docusaurus/docs/overview/feature-model/index.mdx
Comment (danabauer) [reply docusaurus/docs/overview/feature-model/index.mdx:15]: looks good! No need to go to one line

**Review by TristanDiet-TomTom — APPROVED** [review]
Approving, good step forward.

But fully agreeing with the need to restructure afterward.
Names and mechanisms topics are currently a bit floating as children of the introduction and is a mixture of mechanism and feature content pages.
Additionally, looking at it from the perspective of a newcomer looking for an initial narrative in the schema: the current TOC goes:
1. Introduction (~basic welcome)
1.1 Mechanisms/properties that work across Layers (~advanced/expert from schema perspective
2. Layers (~basic from schema perspective)

I would group 1.1 and 2. in one Feature Model bucket with the 1.1 as the last sibling of the 2. topics. This 1.1 subchapter would explain a general introduction on layer crossing properties, the subpages would tackle the respective feature topic pages for names, gers identifier,....
Above is just an initial line of thought, but I would, post-PR, happily discuss additional optimizations.

**Comment by vcschapp** [comment]
Just missing approval from @DavidKarlas. He says he has looked through most and will finish tomorrow.

**Review by DavidKarlas — APPROVED** [review]
Sorry for slow approval.

**Comment by vcschapp** [comment]
Merging to `dev` per review approval status and alignment in this week's Schema TF around "pure docs" PRs.

**Pull request merged**
